06_discord_bot/06_discord_bot.md

```diff
@@ -0,0 +1,219 @@
+
+``` python
+import discord
+import asyncio
+from discord.ext import commands
+bot = commands.Bot(command_prefix='!', description="ce que vous voulez en"
++"description")
+#vous pouvez changer la commande '!' en autre chose si vous le souhaitez
+
+@bot.event
+async def on_ready():
+    print("Logged in as")
+    print(bot.user.name)
+
+#ici figureront les commandes et instructions pour votre bot
+
+bot.run("token") #vous devez ici rentrer le token de votre bot,
+#que vous avez noté plus haut.
+```
+
+Dans l’instruction `bot.run`, vous devez mettre entre guillemets `"` le vrai token de votre bot. Dans tout ce tuto, il figurera en clair sur votre fichier python, mais vous devrez le cacher si vous mettez votre code en ligne un jour
+(rappelez-vous, il doit rester secret).
+
+Les trois premières lignes de ce code indiquent les modules nécessaires, c'est à dire les fichiers dont python a besoin pour comprendre votre code. Par exemple, le module `discord` contient les fonctions créées pour faire le lien entre python et discord.
+Ensuite, en ligne 4, on indique au bot qu'on fera appel à lui grâce au symbole `!` lorsqu'on lui donnera des commandes. Si je veux créer une commande que je veux appeler `coucou`, je lancerai cette commande dans discord en tapant `!coucou`.
+Enfin, la dernière ligne permet de faire le lien entre le code écrit dans le fichier et le bot que vous venez de créer et connecter à discord.
+
+Exécutez votre fichier python contenant le code du Bot. Par exemple si votre fichier s’appelle `Bot.py`, ouvrez un terminal linux, rendez vous dans le dossier contenant votre fichier et tapez `python3 Bot.py`. Vous aurez peut-être besoin de faire une étape intermédiaire si vous rencontrez une erreur. Il faut installer le module `discord` si celui ci n'est pas déjà installé. Pour cela, tapez `pip3 install discord.py`.
+
+Si tout se passe correctement, vous devriez obtenir ce résultat :
+
+![Connexion du bot](ims/im4.png){width=9cm height=1.5cm}
+
+Le terminal vous affiche `Logged in as ...` car dans le fichier que vous venez de lancer se trouve l'instruction
+
+``` python
+@bot.event
+async def on_ready():
+    print("Logged in as")
+    print(bot.user.name)
+```
+
+qui s’exécute au lancement du Bot (et uniquement à ce moment là).
+
+Voilà, votre Bot apparaît désormais en ligne et connecté sur votre
+serveur de test. Mais pour l’instant, il ne fait rien. Nous devons lui
+donner des instructions.
+
+
+
+Une première commande : Hello World
+===================================
+
+Nous allons coder une commande appelée `bonjour` qui demande à notre bot d'envoyer un message `Hello World !`. Dans votre fichier python, à l’endroit où doivent figurer les commandes et instructions (repérées par un commentaire dans votre fichier si vous ne l’avez pas effacé), ajoutez le code suivant :
+
+``` python
+@bot.command(pass_context=True)
+async def bonjour(ctx):
+    msg = ctx.message
+    await msg.channel.send("Hello World !")
```

Comment:
How about using `ctx.send` instead of `ctx.channel.send` (or `ctx.message.channel.send` here)?